lib/Doctrine/DBAL/Query/QueryBuilder.php

```diff
@@ -421,7 +421,7 @@ public function add(string $sqlPartName, $sqlPart, bool $append = false) : self
         $this->state = self::STATE_DIRTY;
 
         if ($append) {
-            if ($sqlPartName === 'orderBy' || $sqlPartName === 'groupBy' || $sqlPartName === 'select' || $sqlPartName === 'set') {
+            if ($sqlPartName === 'orderBy' || $sqlPartName === 'groupBy' || $sqlPartName === 'select' || $sqlPartName === 'set' || $sqlPartName === 'from') {
```

Comment:
Why is this needed?

Reply:
Because of the `|| is_object(...)` introduced right below in the previous PR. This now takes the wrong code path, and the `From` object gets wrapped in an array inside an array. Re-using this code path for the 'from' part does what we expect here.

Anyway, as seen in #3829, I'd like to get rid of this generic `add()` mess ASAP, and leave to each method such as `from()`, `orderBy()` etc. the responsibility to add data to the main array/object in their own way.